pandas/io/gbq.py

```diff
@@ -385,16 +385,25 @@ def run_query(self, query):
         _check_google_client_version()
 
         job_collection = self.service.jobs()
-        job_data = {
-            'configuration': {
-                'query': {
-                    'query': query,
-                    'useLegacySql': self.dialect == 'legacy'
-                    # 'allowLargeResults', 'createDisposition',
-                    # 'preserveNulls', destinationTable, useQueryCache
-                }
+
+        job_config = {
+            'query': {
+                'query': query,
+                'useLegacySql': self.dialect == 'legacy'
+                # 'allowLargeResults', 'createDisposition',
+                # 'preserveNulls', destinationTable, useQueryCache
             }
         }
+        configuration = kwargs.get('configuration')
+        if configuration is not None:
+            if 'query' in configuration:
+                job_config['query'].update(configuration['query'])
+            else:
+                job_config = configuration
```

Comment:
Can you show an example of when `'query'` is not in `configuration`? The unit tests all have `'query'` included in `configuration`.

I think we should ensure that users specify `'query'` in the configuration to prevent users from trying `'load'`, `'extract'` or `'copy'` which are outside of the scope of this PR.

Could we throw an exception or ignore the configuration if `'query'` is not in `configuration`?